text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,2195 @@
+existing protocol. Similarly, a new exporter will be responsible for translating the OTLP messages into this new Arrow-based
+format.
+
+![OTEL Collector](img/0156_OTEL%20-%20OTEL%20Collector.png)
+
+This first step is intended to address the specific use cases of **traffic reduction** and native support of
+**multivariate time-series**. Based on community feedback, many companies want to reduce the cost of transferring
+telemetry data over the Internet. By adding a collector that acts as a point of integration and traffic conversion at
+the edge of a client environment, we can take advantage of the columnar format to eliminate redundant data and optimize
+the compression ratio. This is illustrated in the following diagram.
+
+![Traffic reduction](img/0156_OTEL%20-%20InternetTrafficReduction.png)
+
+> Note 1: A fallback mechanism can be used to handle the case where the new protocol is not supported by the target.
+> More on this mechanism in this [section](#protocol-extension-and-fallback-mechanism).
+
+#### Phase 2
+
+Phase 2 aims to extend the support of Apache Arrow end-to-end and more specifically inside the collector to better
+support the following scenarios: cost efficiency, advanced data processing, data minimization. New receivers, processors,
+and exporters supporting Apache Arrow natively will be developed. A bidirectional adaptation layer OTLP / OTLP Arrow
+will be developed within the collector to continue supporting the existing ecosystem. The following diagram is an
+overview of a collector supporting both OTLP and an end-to-end OTLP Arrow pipeline.
+
+![OTEL Arrow Collector](img/0156_OTEL%20-%20OTELArrowCollector.png)
+
+Implementing an end-to-end column-oriented pipeline will provide many benefits such as:
+
+- **Accelerate stream processing**,
+- **Reduce CPU and Memory usage**,
+- **Improve compression ratio end-to-end**,
+- **Access to the Apache Arrow ecosystem** (query engine, parquet support, ...).
+
+## Protocol Details
+
+The protocol specifications are composed of two parts. The first section describes the new gRPC service supporting
+column-oriented telemetry data. The second section presents the mapping between the OTLP entities and their Apache
+Arrow counterpart.
+
+### EventStream Service
+
+OTLP Arrow defines the columnar encoding of telemetry data and the gRPC-based protocol used to exchange data between
+the client and the server. OTLP Arrow is a bi-directional stream oriented protocol leveraging Apache Arrow for the
+encoding of the telemetry data.
+
+OTLP and OTLP Arrow protocols can be used together and can use the same TCP port. To do so, in addition to the 3
+existing
+services (`MetricsService`, `LogsService` and `TraceService`), we introduce the service `EventsService`
+(see [this protobuf specification](#appendix-a---protocol-buffer-definitions) for more details) exposing a single API endpoint named `EventStream`.
```

Comment:
It is not immediately clear to me why this additional endpoint is offered. Should it be upstreamed into the proto model? 

Event is a loaded term in OpenTelemetry, as log records are also "[events](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/logs#events-and-logs)" and traces have "[events](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#add-events)". Any chance to use a different name here?

Reply:
Indeed it is not the best naming of the century. I'll think about something else, all proposals are welcome.